control-plane/pkg/reconciler/broker/broker_test.go

```diff
@@ -1219,6 +1221,7 @@ func brokerReconciliation(t *testing.T, format string, env config.Env) {
 				finalizerUpdatedEvent,
 			},
 			WantUpdates: []clientgotesting.UpdateActionImpl{
+				SecretFinalizerUpdate("secret-1", SecretFinalizerName),
```

Comment:
Can we create an issue to add a test of the FinalizeKind case? 
Actually, I'd really love to see it in this PR.